composites/Plugin/Shared/components/YoastButton.js

```diff
@@ -39,18 +42,58 @@ export function addButtonStyles( component ) {
 			transform: translateY( 1px );
 			box-shadow: none;
 		}
+
+		// Only needed for Safari 10.
+		span {
+			display: inherit;
+			align-items: inherit;
+			justify-content: inherit;
+			width: 100%;
+		}
 	`;
 }
 
 /**
+ * Returns a component with a button and an inner span element.
+ *
+ * The inner span is only needed to fix a Safari 10 bug with flexbox and button
+ * elements. This bug is fixed in Safari Technology Previs and in the future it
+ * will be possible to remove this component and directly style a button element.
+ * See https://github.com/philipwalton/flexbugs#9-some-html-elements-cant-be-flex-containers
+ *
+ * @returns {ReactElement} The button with inner span.
+ */
+const YoastButtonBase = ( { className, onClick, type, children } ) => (
+	<button className={ className } onClick={ onClick } type={ type }>
+		<span>
+			{ children }
+		</span>
+	</button>
+);
+
+YoastButtonBase.propTypes = {
+	className: PropTypes.string,
+	onClick: PropTypes.func,
+	type: PropTypes.string,
+	children: PropTypes.string,
+};
+
+YoastButtonBase.defaultProps = {
+	type: "button",
+};
+
+/**
  * Returns a Button with the Yoast button style.
  *
+ * We're styling the YoastButtonBase component just because of the Safari 10 bug.
```

Comment:
Maybe create a reference to the bug you mean (e.g. `(see YoastButtonBase JSDoc)`)